### PR TITLE
Introduce kubernetes client cache to improve API call response times

### LINKED ISSUE #115: Assess and Fix API Slowness
The API feels very slow when working with many objects (see integration in VSHN Portal). While this is at first sight a cosmetic issue, it hinders to have a good user feeling.

## Steps to Reproduce the Problem

1. Login to control.vshn.net
1. Browse around in Lieutenant section
1. Feel the slowness

## Actual Behavior

It's very slow, also reported by the Portal itself.

## Expected Behavior

It's fast.

### PR DESCRIPTION
**Pull request by ccremer**
* Updates all dependencies as far as possible
* Fixes #115 
* Adds a LRU cache that caches Kubernetes client instances by tokens. After 128 cached tokens (and client instances), the least recently used client will be evicted. The first call with an uncached token takes about 2s due to the client instantiation. This was deemed to be an acceptable approach as discussed in #115 without needing to refactor a major part.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Link this PR to related issues.

### CODE REVIEW

**Review by simu — APPROVED** [review]

File: pkg/service/kubernetes_auth.go [hunk 1]
```diff
@@ -25,18 +26,22 @@ var (
 	}
 )
 
-func init() {
-	synv1alpha1.SchemeBuilder.AddToScheme(scheme.Scheme)
+func createCache() *lruCache.Cache {
+	cache, err := lruCache.NewWithEvict(128, nil)
```
Comment (srueg) [inline pkg/service/kubernetes_auth.go:30]: Would it make sense to make the size configurable via env var?

File: pkg/service/kubernetes_auth.go [hunk 2]
```diff
@@ -60,14 +65,20 @@ func (k *KubernetesAuth) JWTAuth(next echo.HandlerFunc) echo.HandlerFunc {
...
+			if err != nil {
+				return err
+			}
+			k.cache.Add(token, cachedClient)
 		}
 
 		apiContext := &APIContext{
 			Context: c,
-			client:  client,
+			context: context.TODO(),
```
Comment (srueg) [inline pkg/service/kubernetes_auth.go:80]: Is this context used somewhere?
I think we can remove it from the `APIContext` struct.

Comment (ccremer) [reply pkg/service/kubernetes_auth.go:80]: Yes, it's used in all client.List/Get/Create calls etc.

File: pkg/service/kubernetes_auth.go (hunk 1 above)
Comment (ccremer) [reply pkg/service/kubernetes_auth.go]: added. WDYT now?

File: pkg/service/kubernetes_auth.go (hunk 2 above)
Comment (ccremer) [reply pkg/service/kubernetes_auth.go:95]: The alternative would be to set `context.TODO()` for these calls individually and not store it in the APIContext property

Comment (srueg) [reply pkg/service/kubernetes_auth.go:95]: Hmm I see. I guess we should use `c.Request().Context()` instead...

But I'm fine with doing it in another PR...

**Review by srueg — APPROVED** [review]

File: pkg/service/kubernetes_auth.go (hunk 2 above)
Comment (ccremer) [reply pkg/service/kubernetes_auth.go]: ah, was searching for something like this, didn't know it's hidden in Request().

A bit of search/replace was easy.

**Pull request merged**